Sentinel/Classes/CrashDetection/Internal/Handlers/CrashHandler.swift

```diff
@@ -0,0 +1,60 @@
+//
+//  CrashHandler.swift
+//  MasterConnect
+//
+//  Created by Zvonimir Medak on 11.04.2024..
+//  Copyright © 2024 Signify. All rights reserved.
+//
+
+// Inspiration link: https://github.com/DebugSwift/DebugSwift
+
+import Foundation
+
+final class CrashHandler {
+
+    // MARK: - Singleton -
+
+    static let shared = CrashHandler()
+
+    // MARK: - Private properties -
+
+    private let uncaughtExceptionHandler: CrashUncaughtExceptionHandler
+    private let signalExceptionHandler: CrashSignalExceptionHandler
+    private var exceptionReceiveClosure: ((Int32?, NSException?, String) -> Void)?
+
+    // MARK: - Init -
+
+    init() {
+        let signalHandler = SignalPrehandlerManager()
+        uncaughtExceptionHandler = CrashUncaughtExceptionHandler(preHandlers: signalHandler.preHandlers)
+        signalExceptionHandler = CrashSignalExceptionHandler(preHandlers: signalHandler.preHandlers)
+
+        CrashUncaughtExceptionHandler.exceptionReceiveClosure = { [weak self] signal, exception, info in
+            self?.exceptionReceiveClosure?(signal, exception, info)
+            CrashManager.save(
+                crash: .init(type: .nsexception, details: .builder(name: info))
+            )
```

Comment:
Even though I'm fine with this, I would rather "register" or "add" result handler and then call it here instead of directly using this manager.

Reason; if we want to use another way of saving or previewing data for the user in the future (e.g., saving it to the database).

Same goes for the signal exception handler couple of lines below.